controllers/common/handlerWithDiff.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2021 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package common
+
+import (
+	"context"
+
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+
+	policiesv1beta1 "open-cluster-management.io/governance-policy-propagator/api/v1beta1"
+)
+
+var _ handler.EventHandler = &enqueueRequestsFromMapFuncWithDiff{}
+
+// EnqueueRequestsFromMapFunc same as original EnqueueRequestsFromMapFunc
+// execept this only queues the diff between the new and old objects
+func EnqueueRequestsFromMapFuncWithDiff(fn handler.MapFunc) handler.EventHandler {
+	return &enqueueRequestsFromMapFuncWithDiff{
+		toRequests: fn,
+	}
+}
+
+// EnqueueRequestsFromMapFunc same as original EnqueueRequestsFromMapFunc
+// execept this doesn't queue old object for update
+type enqueueRequestsFromMapFuncWithDiff struct {
+	// Mapper transforms the argument into a slice of keys to be reconciled
+	toRequests handler.MapFunc
+}
+
+// Create implements EventHandler
+func (e *enqueueRequestsFromMapFuncWithDiff) Create(ctx context.Context, evt event.CreateEvent,
+	q workqueue.RateLimitingInterface,
+) {
+	e.mapAndEnqueue(ctx, q, evt.Object)
+}
+
+// Update implements EventHandler
+// Enqueues the diff between the new and old policy sets in the UpdateEvent
+func (e *enqueueRequestsFromMapFuncWithDiff) Update(ctx context.Context, evt event.UpdateEvent,
+	q workqueue.RateLimitingInterface,
+) {
+	//nolint:forcetypeassert
+	newPolicySet := evt.ObjectNew.(*policiesv1beta1.PolicySet)
+	//nolint:forcetypeassert
+	oldPolicySet := evt.ObjectOld.(*policiesv1beta1.PolicySet)
+
+	newPoliciesMap := make(map[policiesv1beta1.NonEmptyString]bool)
+	diffPolicies := []policiesv1beta1.NonEmptyString{}
+
+	for _, plc := range newPolicySet.Spec.Policies {
+		newPoliciesMap[plc] = true
+	}
+
+	for _, plc := range oldPolicySet.Spec.Policies {
+		if !newPoliciesMap[plc] {
+			diffPolicies = append(diffPolicies, plc)
+		}
+	}
+
+	diffPolicySet := policiesv1beta1.PolicySet{Spec: policiesv1beta1.PolicySetSpec{
+		Policies: diffPolicies,
```

Comment:
If I'm reading this correctly, `diffPolicies` only contains policies that were previously in the `PolicySet` where as it should contain policies that are no longer in the old `PolicySet` and the new policies in the new `PolicySet`.